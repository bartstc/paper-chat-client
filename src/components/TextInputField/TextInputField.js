import React from 'react';
import PropTypes from 'prop-types';
import { Input, Label } from './TextInputField.styles';

const TextInputField = ({
  name,
  id,
  placeholder,
  value,
  label,
  error,
  type,
  onChange
}) => {
  return (
    <>
      <Label htmlFor={id}>{label}</Label>
      <Input
        type={type}
        id={id}
        error={error}
        placeholder={placeholder}
        name={name}
        value={value}
        onChange={onChange}
      />
    </>
  );
};

TextInputField.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  label: PropTypes.string,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

TextInputField.defaultProps = {
  type: 'text'
};

export default TextInputField;
