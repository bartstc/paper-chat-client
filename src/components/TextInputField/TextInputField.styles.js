import styled from 'styled-components';
import { color } from '../../utils/styles';

export const Label = styled.label`
  font-size: 0.9rem;
  color: ${color.accent};
  font-weight: 300;
  margin-bottom: 0.4em;
  width: 100%;
  margin-top: 1.5em;
`;

export const Input = styled.input`
  width: 100%;
  height: 40px;
  padding-left: 0.8em;
  border: ${props =>
    props.error ? '1px solid red' : '1px solid' + color.accent};
  border-radius: 8px;
  font-size: 1rem;

  &::placeholder {
    color: #333;
    font-size: 1rem;
  }
`;
