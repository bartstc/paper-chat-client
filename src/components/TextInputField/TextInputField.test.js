import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextInputField from './TextInputField';

describe('<TextInputField />', () => {
  const mockOnChange = jest.fn();

  const props = {
    id: 'username',
    placeholder: 'username',
    name: 'username',
    value: 'John Doe',
    label: 'Enter username',
    onChange: mockOnChange
  };

  let wrapper = mount(<TextInputField {...props} />);

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('label', () => {
    const label = wrapper.find('label');

    it('exists', () => {
      expect(label).toHaveLength(1);
    });

    it('contains correct htmlFor and text value', () => {
      expect(label.prop('htmlFor')).toEqual('username');
      expect(label.text()).toEqual('Enter username');
    });
  });

  describe('input', () => {
    const input = wrapper.find('input');

    it('exists', () => {
      expect(input).toHaveLength(1);
    });

    it('contains correct props', () => {
      expect(input.prop('type')).toEqual('text');
      expect(input.prop('name')).toEqual('username');
      expect(input.prop('id')).toEqual('username');
      expect(input.prop('placeholder')).toEqual('username');
    });
  });
});
