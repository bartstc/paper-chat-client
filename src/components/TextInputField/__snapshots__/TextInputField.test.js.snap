// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputField /> renders correctly 1`] = `
<TextInputField
  id="username"
  label="Enter username"
  name="username"
  onChange={[MockFunction]}
  placeholder="username"
  type="text"
  value="John Doe"
>
  <styled.label
    htmlFor="username"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "EdmCa",
            "rules": Array [
              "
  font-size: 0.9rem;
  color: ",
              "#0075C4",
              ";
  font-weight: 300;
  margin-bottom: 0.4em;
  width: 100%;
  margin-top: 1.5em;
",
            ],
          },
          "displayName": "styled.label",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      htmlFor="username"
    >
      <label
        className="sc-bdVaJa EdmCa"
        htmlFor="username"
      >
        Enter username
      </label>
    </StyledComponent>
  </styled.label>
  <styled.input
    id="username"
    name="username"
    onChange={[MockFunction]}
    placeholder="username"
    type="text"
    value="John Doe"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "UGWtj",
            "rules": Array [
              "
  width: 100%;
  height: 40px;
  padding-left: 0.8em;
  border: ",
              [Function],
              ";
  border-radius: 8px;
  font-size: 1rem;

  &::placeholder {
    color: #333;
    font-size: 1rem;
  }
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      id="username"
      name="username"
      onChange={[MockFunction]}
      placeholder="username"
      type="text"
      value="John Doe"
    >
      <input
        className="sc-bwzfXH UGWtj"
        id="username"
        name="username"
        onChange={[MockFunction]}
        placeholder="username"
        type="text"
        value="John Doe"
      />
    </StyledComponent>
  </styled.input>
</TextInputField>
`;
